apps/modernization-ui/src/apps/page-builder/pages/BusinessRulesLibrary/Add/AddBusinessRule.tsx

```diff
@@ -0,0 +1,199 @@
+import { Button, ButtonGroup, Form, Grid, Icon } from '@trussworks/react-uswds';
+import { useEffect, useState } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+import { useNavigate, useParams } from 'react-router-dom';
+import './AddBusinessRule.scss';
+import { PageRuleControllerService, ViewRuleResponse } from '../../../generated';
+import { useAlert } from 'alert';
+import BusinessRulesForm from '../BusinessRulesForm';
+import { PageBuilder } from '../../PageBuilder/PageBuilder';
+import { Breadcrumb } from 'breadcrumb';
+import { authorization } from 'authorization';
+
+export type FormValues = {
+    ruleFunction: string;
+    sourceIdentifier: string;
+    sourceText: string;
+    comparator?: string;
+    sourceValueIds?: string[];
+    sourceValueText?: string[];
+    sourceValue?: {
+        sourceValueId?: string[];
+        sourceValueText?: string[];
+    };
+    targetValueIdentifier?: string[];
+    targetType: string;
+    ruleDescription: string;
+    targetValueText?: string[];
+    anySourceValue?: boolean;
+};
+
+const AddBusinessRule = () => {
+    const navigate = useNavigate();
+    const form = useForm<FormValues>({
+        defaultValues: { targetType: 'SUBSECTION', anySourceValue: false },
+        mode: 'onChange'
+    });
+    const [selectedFieldType, setSelectedFieldType] = useState('');
+    const { pageId, ruleId } = useParams();
+    const { showAlert } = useAlert();
+
+    useEffect(() => {
+        if (ruleId) {
+            PageRuleControllerService.viewRuleResponseUsingGet({
+                authorization: authorization(),
+                ruleId: Number(ruleId)
+            }).then((resp: ViewRuleResponse) => {
+                form.setValue('anySourceValue', resp?.anySourceValue!);
+                form.setValue('ruleDescription', resp?.ruleDescription!);
+                form.setValue('ruleFunction', resp?.ruleFunction!);
+                form.setValue('comparator', resp?.comparator!);
+                form.setValue('targetType', resp?.targetType!);
+                form.setValue('sourceIdentifier', resp?.sourceIdentifier!);
+                form.setValue('sourceText', resp?.sourceIdentifier!);
+                form.setValue('sourceValueIds', resp?.sourceValue?.sourceValueId!);
+                form.setValue('sourceValueText', resp?.sourceValue?.sourceValueText!);
+                form.setValue('targetValueIdentifier', resp?.targetValueIdentifier!);
+                form.setValue('targetValueText', ['testing']);
```

Comment:
You might be able to just pass an object into the `form.reset` method instead of calling `form.setValue` a bunch of times